common/scala/src/main/scala/org/apache/openwhisk/common/Logging.scala

```diff
@@ -183,12 +183,11 @@ private object Emitter {
  * @param subAction more specific identifier for "action", like `runc.resume`
  * @param tags tags can be used for whatever granularity you might need.
  */
-case class LogMarkerToken(
-  component: String,
-  action: String,
-  state: String,
-  subAction: Option[String] = None,
-  tags: Map[String, String] = Map.empty)(measurementUnit: MeasurementUnit = MeasurementUnit.none) {
+case class LogMarkerToken(component: String,
+                          action: String,
+                          state: String,
+                          subAction: Option[String] = None,
+                          tags: TagSet = TagSet.Empty)(measurementUnit: MeasurementUnit = MeasurementUnit.none) {
```

Comment:
So far most of . the code is not using Kamon api. Would it be possible to keep `Map` here and internally convert it to `TagSet` (via `TagSet.from`). That would avoid modifying all the places where token is used

Reply:
Good call, updated!